### PR TITLE
Homework - Database-week1

### PR DESCRIPTION
**Pull request by halynabondar**
Homework
I have designed the following queries
1. Find out how many tasks are in the task table
2. Find out how many tasks in the task table do not have a valid due date
3. Find all the tasks that are marked as done
4. Find all the tasks that are not marked as done
5. Get all the tasks, sorted with the most recently created first
6. Get the single most recently created task
7. Get the title and due date of all tasks where the title or description contains database
8. Get the title and status (as text) of all tasks
9. Get the name of each status, along with a count of how many tasks have that status
10. Get the names of all statuses, sorted by the status with most tasks first

Class exercises
Implemented the following queries:

1. Select the names and phones of all users;
2. Select the name of the user with id=10;
3. Find how many users exist in the database;
4. Select the names of the first 5 users in the database;
5. Select the names of the last 3 users in the database;
6. Sum all the ids in the user table;
7. Select all users and order them alphabetically by name;
8. Find all tasks that include SQL either on the title or on the description;
9. Find the title of all tasks that the user Maryrose is responsible for;
10. Find how many tasks each user is responsible for;
11. Find how many tasks with a status=Done each user is responsible for;

### CODE REVIEW

File: Database/DB-week1/Homework.sql [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+-- 2. Find out how many tasks in the task table do not have a valid due date
+SELECT COUNT(*)
+FROM task
+WHERE due_date IS NULL;
+
+-- 3. Find all the tasks that are marked as done
+SELECT *
+FROM task
+WHERE status_id = 3;
```
Comment (jpganz) [inline Database/DB-week1/Homework.sql]: what if done is not status 3?

File: Database/DB-week1/Homework.sql [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+LIMIT 1;
+
+-- 7. Get the title and due date of all tasks where the title or description contains database
+SELECT title,
+       due_date
+FROM task
+WHERE title LIKE '%database%'
+   OR description LIKE '%database%';
+
+-- 8. Get the title and status (as text) of all tasks
```
Comment (jpganz) [inline Database/DB-week1/Homework.sql:39]: this query is wrong, please take a second look

File: Database/DB-week1/Homework.sql [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+SELECT title,
+       status_id
+FROM task;
+
+-- 9. Get the name of each status, along with a count of how many tasks have that status
+SELECT status_id, COUNT(*) AS task_count
```
Comment (jpganz) [inline Database/DB-week1/Homework.sql]: this query is almost there, please check it again

File: Database/DB-week1/Homework.sql [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+FROM task
+GROUP BY status_id;
+
+-- 10. Get the names of all statuses, sorted by the status with most tasks first
+SELECT status_id, COUNT(*) AS task_count
```
Comment (jpganz) [inline Database/DB-week1/Homework.sql]: same as previous

File: Database/DB-week1/Homework.sql [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+SELECT name,
+       phone
+FROM user;
+
+-- Select the name of the user with id=10;
+SELECT name
+FROM user
+WHERE id = 10;
+
+-- Find how many users exist in the database;
```
Comment (jpganz) [inline Database/DB-week1/Homework.sql:72]: this was from the user table :D

File: Database/DB-week1/Homework.sql [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+SELECT COUNT(*) AS user_count
+FROM mysql.user;
+
+-- Select the names of the first 5 users in the database;
+SELECT name
+FROM user
+LIMIT 5;
+
+-- Select the names of the last 3 users in the database;
```
Comment (jpganz) [inline Database/DB-week1/Homework.sql:81]: this query is wrong, please take a second look

**Review by jpganz — CHANGES_REQUESTED** [review]

File: Database/DB-week1/Homework.sql (hunk 2 above)
Comment (halynabondar) [reply Database/DB-week1/Homework.sql:39]: the query works just fine and I can see the results containing 'database', could you be more specific what exactly is wrong in this query?
image

**Review by jpganz — APPROVED** [review]

**Pull request merged**